src/Net/Http.fs

```diff
@@ -386,24 +1017,105 @@ module private HttpHelpers =
         let buffer = Array.zeroCreate (4 * 1024)
         let output = new MemoryStream()
         let reading = ref true
-      
+
         while reading.Value do
             // Download one (at most) 4kb chunk and copy it
             let! count = stream.AsyncRead(buffer, 0, buffer.Length)
             output.Write(buffer, 0, count)
             reading := count > 0
-      
-        output.Seek(0L, SeekOrigin.Begin) |> ignore 
-        return output 
+
+        output.Seek(0L, SeekOrigin.Begin) |> ignore
+        return output
     }
 
-    let getProperty (typ:Type) obj prop =        
+    /// A stream class that abstracts away writing the contents of a series of other streams, closing them as they are consumed.  Non-seekable, reading-only stream.
+    type CombinedStream(length, streams: Stream seq) =
+        inherit Stream() with
+            let mutable v = 0L
+            let mutable streams = streams |> Seq.cache
+
+            let rec readFromStream buffer offset count =
+                match streams |> Seq.isEmpty with
+                | true ->
+                    0
+                | false ->
+                    let stream = Seq.head streams
```

Comment:
Since it's a boolean maybe just a simple if?

                if Seq.isEmpty streams then
                    0
                else

Reply:
Agreed